vortex-datafusion/src/persistent/format.rs

```diff
@@ -75,14 +77,21 @@ impl FileFormat for VortexFormat {
         store: &Arc<dyn ObjectStore>,
         objects: &[ObjectMeta],
     ) -> DFResult<SchemaRef> {
-        let mut file_schemas = Vec::default();
-        for o in objects {
-            let initial_read = self.initial_read_cache.try_get(o, store.clone()).await?;
-
-            let lazy_dtype = initial_read.lazy_dtype();
-            let s = infer_schema(lazy_dtype.value()?)?;
-            file_schemas.push(s);
-        }
+        let file_schemas = stream::iter(objects.iter().cloned())
+            .map(|o| {
+                let store = store.clone();
+                let cache = self.initial_read_cache.clone();
+                async move {
+                    let initial_read = cache.try_get(&o, store).await?;
+                    let lazy_dtype = initial_read.lazy_dtype();
+                    let s = infer_schema(lazy_dtype.value()?)?;
+
+                    VortexResult::Ok(s)
+                }
+            })
+            .buffered(16)
```

Comment:
I think 64, just because this is super I/O bound. And can you make this a field in the VortexFormat struct?